dvc/repo/metrics/show.py

```diff
@@ -70,7 +70,9 @@ def _extract_metrics(metrics, path, rev):
 
 @error_handler
 def _read_metric(path, fs, rev, **kwargs):
-    val = load_yaml(path, fs=fs)
+    suffix = fs.path.suffix(path).lower()
+    loader = LOADERS[suffix]
+    val = loader(path, fs=fs)
```

Comment:
Yes, It'd be nice to refactor this. But there's two main ways the suffix is obtained: `fs.path.suffix(fs_path)` and `os.path.splitext(fs_path)`. Is it safe to assume they are and always will be equivalent? Should one be preferred over the other?

Reply:
> Is it safe to assume they are and always will be equivalent? Should one be preferred over the other?

I think it would always be the same. Anyhow, I think we can leave that out of this P.R.